### ISSUE TITLE
Per-post type events fire fire for all types of posts

### ISSUE THREAD

**Issue by JDGrimes** [WordPoints/hooks-api#155]
While working on #23, I discovered that if there is a `post_publish\post` reaction and a `post_publish\page` reaction, both of them will fire when a post of any type is published.  This is because we only use the namespace part of the arg (`post`) when retrieving the arg value from the action, rather than the whole thing (`post\{post_type}`). (Actually, we check with the full arg name first, but fall back to the namespace only when we don't find that.)

In short, we never check that the post type matches the dynamic portion of the slug.

**Comment by JDGrimes** [comment]
> Per-post type events fire fire [sic] for all types of posts

Exactly.

**Comment by JDGrimes** [comment]
Part of the problem here is that we check whether the action as a whole should fire, but not whether each individual event should be triggered. This means that we really don't have a good way to even fix this issue. That is, unless we force the action to be registered per-post type just as the events and args are.

**Comment by JDGrimes** [comment]
> That is, unless we force the action to be registered per-post type just as the events and args are.

That kind of seems silly though, because then we'd have to loop through all of those actions. I guess that really it isn't any different than what we're doing now though in that regard, because we already have to loop through all of those events.

Registering the actions per-post type would also give us more granularity in regard to other things too. We can reuse the class of course.

**Comment by JDGrimes** [comment]
However, I guess that is really a separate issue from whether we want to allow individual events to be aborted (and maybe this ticket ain't big enough for the two of 'em).

**Comment by JDGrimes** [comment]
I suppose one option here would be to just build handling for namespaced actions like this right into the router. We could check if the action name had a dynamic portion. But that would require us to register the actions per-post type anyway, which would make the whole point moot.

If we wanted to build this in, we'd need to introduce a way to check if the action was generic, I guess. And then we'd need to be able to get the dynamic portion of the action arg name (the post type in this case) and compare that against each event slug.

**Comment by JDGrimes** [comment]
However, that last suggestion wouldn't offer us the same granularity that we'd have by just registering the actions per post type.

**Comment by JDGrimes** [comment]
The only real benefit of not registering the actions per-post type, and the only thing holding me back from just doing it, is that it takes slightly more memory since the actions and args are registered multiple times instead of just once. This is also one of the main reasons that we did this this way to start with.

However, when we really consider it, there are never going to be so many post types and actions that this causes a problem. Much more memory is probably consumed just by the definitions of each post type and their related paraphernalia than by our actions. So while this was a nice optimization, we already split the events and entities, so I see no reason not to proceed with splitting the actions too.
